preview_generator/preview/builder_factory.py

```diff
@@ -38,22 +42,36 @@ def get_preview_builder(
         try:
             return self._builder_classes[mimetype]()  # nopep8 get class and instantiate it
         except KeyError:
-            raise UnsupportedMimeType('Unsupported mimetype: {}'.format(mimetype))
+            raise UnsupportedMimeType(
+                'Unsupported mimetype: {}'.format(mimetype)
+            )
 
     def get_file_mimetype(self, file_path: str, file_ext: str='') -> str:
         """
         return the mimetype of the file. see python module mimetype
         """
-        str, encoding = mimetypes.guess_type(file_path, strict=False)
-        if not str or str == 'application/octet-stream':
+        _str, encoding = mimetypes.guess_type(file_path, strict=False)
+        if not _str or _str == 'application/octet-stream':
             mime = magic.Magic(mime=True)
-            str = mime.from_file(file_path)
-
-        if not str or str == 'application/octet-stream':
+            _str = mime.from_file(file_path)
+
+        if _str and (_str in ['text/xml', 'text/plain']):
+            raw_mime = Popen(
+                ['mimetype', file_path],
+                stdin=PIPE, stdout=PIPE, stderr=PIPE
+            ).communicate()[0]
+            _str = (
+                raw_mime
+                .decode("utf-8")
+                .replace(file_path, '')
+                .replace(': ', '')
+                .replace('\n', '')
+            )
+        if not _str or _str == 'application/octet-stream':
             complete_path = file_path + '.' + file_ext
-            str, encoding = mimetypes.guess_type(complete_path)
+            _str, encoding = mimetypes.guess_type(complete_path)
```

Comment:
If y have a good reading of PEP8:

```
single_trailing_underscore_: used by convention to avoid conflicts with Python keyword, e.g.
```

You should name `str_` instead `_str`.